prog/weaveutil/attach.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"syscall"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/vishvananda/netns"
+
+	weavenet "github.com/weaveworks/weave/net"
+)
+
+func attach(args []string) error {
+	if len(args) < 4 {
+		cmdUsage("attach-container", "[--no-multicast-route] <container-id> <bridge-name> <mtu> <cidr>...")
+	}
+
+	withMulticastRoute := true
+	if args[0] == "--no-multicast-route" {
+		withMulticastRoute = false
+		args = args[1:]
+	}
+
+	pid, err := findPid(args[0])
+	if err != nil {
+		return err
```

Comment:
Have you thought about wrapping each returned err? E.g. `return fmt.Errorf("findPid failed: %s", err)`. It might be a bit redundant, but IMHO it would improve debugging.